grype/db/v6/blobs.go

```diff
@@ -0,0 +1,105 @@
+	SeveritySchemeCVSSV4 SeverityScheme = "CVSSv4"
+
+	// SeveritySchemeHML is a string severity scheme (High, Medium, Low)
+	SeveritySchemeHML SeverityScheme = "HML"
+
+	// SeveritySchemeCHMLN is a string severity scheme (Critical, High, Medium, Low, Negligible)
+	SeveritySchemeCHMLN SeverityScheme = "CHMLN"
+)
+
+// VulnerabilityBlob represents the core advisory record for a single known vulnerability from a specific provider.
+type VulnerabilityBlob struct {
+	// ID is the lowercase unique string identifier for the vulnerability relative to the provider
+	ID string `json:"id"`
+
+	// ProviderName of the Vunnel provider (or sub processor responsible for data records from a single specific source, e.g. "ubuntu")
+	ProviderName string `json:"provider"`
+
+	// Assigner is a list of names, email, or organizations who submitted the vulnerability
+	Assigner []string `json:"assigner,omitempty"`
+
+	// Status conveys the actionability of the current record
+	Status VulnerabilityStatus `json:"status"`
+
+	// Description of the vulnerability as provided by the source
+	Description string `json:"description"`
+
+	// PublishedDate is the date the vulnerability record was first published
+	PublishedDate *time.Time `json:"published,omitempty"`
+
+	// ModifiedDate is the date the vulnerability record was last modified
+	ModifiedDate *time.Time `json:"modified,omitempty"`
+
+	// WithdrawnDate is the date the vulnerability record was withdrawn
+	WithdrawnDate *time.Time `json:"withdrawn,omitempty"`
+
+	// References are URLs to external resources that provide more information about the vulnerability
+	References []Reference `json:"refs,omitempty"`
+
+	// Aliases is a list of IDs of the same vulnerability in other databases, in the form of the ID field. This allows one database to claim that its own entry describes the same vulnerability as one or more entries in other databases.
+	Aliases []string `json:"aliases,omitempty"`
+
+	// Severities is a list of severity indications (quantitative or qualitative) for the vulnerability
+	Severities []Severity `json:"severities,omitempty"`
+}
+
+// Reference represents a single external URL and string tags to use for organizational purposes
+type Reference struct {
+	// URL is the external resource
+	URL string `json:"url"`
+
+	// Tags is a free-form organizational field to convey additional information about the reference
+	Tags []string `json:"tags,omitempty"`
+}
+
+// Severity represents a single string severity record for a vulnerability record
+type Severity struct {
+	// Scheme describes the quantitative method used to determine the Score, such as "CVSS_V3". Alternatively this makes
+	// claim that Value is qualitative, for example "HML" (High, Medium, Low), CHMLN (critical-high-medium-low-negligible)
+	Scheme SeverityScheme `json:"scheme"`
+
+	// Value is the severity score (e.g. "7.5", "CVSS:4.0/AV:N/AC:L/AT:N/PR:H/UI:N/VC:L/VI:L/VA:N/SC:N/SI:N/SA:N",  or "high" )
+	Value string `json:"value"`
```

Comment:
I think I one point we had this as `Score` and then the shape of the object could be determined by the `Scheme` because some scoring mechanisms might not be represented as a single string?  But maybe it was determined that was too complex later, I can't remember back that far now

Reply:
Maybe it was that if we want to represent something more complex than a string value here it would be a blob lookup to another table?

Reply:
Ah, yeah I think you had moved something like future EPSS over to its own table, so I guess it would be a similar approach for any new measure that has something more complex than a string as a representation

Reply:
well, what I could do is make value an `any` type so that we could put in various shapes here. I can see this for score/vector for CVSS (even though we could calculate the score out from the vector... but why do that work again if we already have the value).